src/main/java/com/encora/BinarySearchTree.java

```diff
@@ -137,24 +137,27 @@ public int getRoot() {
 
     @Override
     public String toString() {
-        if (root != null) {
-            if (root.getLeft() != null) {
-                if (root.getLeft().getLeft() != null) {
-                    return toString();
-                }
-                System.out.println(root.getLeft().getValue());
-            }
-        }
-        if (root != null) {
-            System.out.println(root.getValue());
-        }
-        if (root.getRight() != null) {
-            if (root.getRight().getRight() != null) {
-                return toString();
-            }
-            System.out.println(root.getRight().getValue());
+        return inOrder().toString();
+    }
+
+    public String toStringOpt(String opt) {
```

Comment:
What about adding an enum to manage the different options? Also, the idea of having `toString` implemented was to override the original from Object class. For me would make more sense to get rid off of this and trust on the `inOrder`, `preOrder` and `postOrder` implementations.